src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/mailer.html.twig

```diff
@@ -101,135 +63,202 @@
         </div>
     {% endif %}
 
-    {% for transport in events.transports %}
-        <div class="card-block">
+    {% if events.transports|length > 1 %}
+        {% for transport in events.transports %}
+            <h2><code>{{ transport }}</code> transport</h2>
+            {{ _self.render_transport_details(collector, transport) }}
+        {% endfor %}
+    {% else %}
+        {{ _self.render_transport_details(collector, events.transports|first, true) }}
+    {% endif %}
+
+    {% macro render_transport_details(collector, transport, show_transport_name = false) %}
+        <div class="card">
+            {% set num_emails = collector.events.events(transport)|length %}
+            {% if num_emails > 1 %}
+                <div class="mailer-email-summary-table-wrapper">
+                    <table class="mailer-email-summary-table">
+                        <thead>
+                            <tr>
+                                <th>#</th>
+                                <th>Subject</th>
+                                <th>To</th>
+                                <th class="visually-hidden">Actions</th>
+                            </tr>
+                        </thead>
+                        <tbody>
+                            {% for event in collector.events.events(transport) %}
+                                <tr class="mailer-email-summary-table-row {{ loop.first ? 'active' }}" data-target="#email-{{ loop.index }}">
+                                    <td>{{ loop.index }}</td>
+                                    <td>{{ event.message.headers.get('subject').bodyAsString() ?? '(No subject)' }}</td>
+                                    <td>{{ (event.message.headers.get('to').bodyAsString() ?? '(empty)')|replace({'To:': ''}) }}</td>
+                                    <td class="visually-hidden"><button class="mailer-email-summary-table-row-button" data-target="#email-{{ loop.index }}">View email details</button></td>
```

Comment:
i don't know the feasibility but can it be swapped to use default html details https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details
(to avoid javascript)

Reply:
this won't work. It  can only expand details inside the cell of the table. That's not what we want in this design.